libraries/helpers.rb

```diff
@@ -17,155 +17,101 @@
 
 module MarketplaceImageCookbook
   module Helpers
-    def install_marketplace
-      chef_ingredient 'chef-marketplace' do
-        version new_resource.marketplace_version
-        config marketplace_config
-        action :upgrade
-      end
-
-      ingredient_config 'chef-marketplace' do
-        notifies :reconfigure, 'chef_ingredient[chef-marketplace]', :immediately
-      end
+    def marketplace_products
+      aws_products + azure_products + gce_products
     end
 
-    def install_server
-      chef_ingredient 'chef-server' do
-        config server_config
-        version new_resource.server_version
-        action :upgrade
-      end
-
-      ingredient_config 'chef-server'
-
-      chef_ingredient 'reporting' do
-        config new_resource.reporting_config
-        version new_resource.reporting_version
-        action :upgrade
-      end
-
-      ingredient_config 'reporting'
-
-      chef_ingredient 'manage' do
-        config manage_config
-        version new_resource.manage_version
-        action :upgrade
-      end
-
-      ingredient_config 'manage'
+    def enabled_products
+      enabled_aws_products + enabled_azure_products + enabled_gce_products
     end
 
-    def install_analytics
-      chef_ingredient 'analytics' do
-        config new_resource.analytics_config
-        version new_resource.analytics_version
-        action :upgrade
-      end
-
-      ingredient_config 'analytics'
+    def enabled_builders
+      enabled_products.map { |p| p['name'] }
     end
 
-    def install_compliance
-      chef_ingredient 'compliance' do
-        config compliance_config
-        version new_resource.compliance_version
-        action :upgrade
-      end
-
-      ingredient_config 'compliance'
+    def aws_products
+      node['marketplace_image']['aws']['public']['aio']['products'] +
+        node['marketplace_image']['aws']['public']['compliance']['products'] +
+        node['marketplace_image']['aws']['ic']['aio']['products'] +
+        node['marketplace_image']['aws']['ic']['compliance']['products'] +
+        [
+          node['marketplace_image']['aws']['public']['aio']['fcp'],
+          node['marketplace_image']['aws']['public']['compliance']['fcp']
+        ]
     end
 
-    def install_aio
-      install_server
-      install_analytics
+    def enabled_aws_builders
+      enabled_aws_products.map { |p| p['name'] }
     end
 
-    def uninstall_server
-      chef_ingredient 'chef-server' do
-        action :uninstall
-      end
-
-      chef_ingredient 'reporting' do
-        action :uninstall
-      end
-
-      chef_ingredient 'manage' do
-        action :uninstall
-      end
+    def aws_builders
+      aws_products.map { |p| p['name'] }
     end
 
-    def uninstall_marketplace
-      chef_ingredient 'chef-marketplace' do
-        action :uninstall
-      end
+    def enabled_aws_products
```

Comment:
I feel like a lot of these helpers could be made more dynamic, looping over known sets of clouds, marketplaces, etc. so when it comes time to add another marketplace, we just add it to the list vs. having to create 4-5 more helpers that are relatively static.

Not something we need to do for this version given we don't know what the growth plan looks like, but I could see this as a monotonous unnecessary task to add these helpers :)